File: Network/src/Operation.java

public class Operation 
{
	private Dispositive from, to;
	private int from2, to2;
	private String message;
	private boolean broadcast;
	
	public Operation(Dispositive f, Dispositive t, String m, int f2, int t2, boolean b)
	{
		from=f;
		to=t;
		message=m;
		from2=f2;
		to2=t2;
		broadcast=b;
	}

	public Dispositive getTo() 
	{
		return to;
	}
	
	public Dispositive getFrom() 
	{
		return from;
	}
	
	public int getTo2() 
	{
		return to2;
	}
	
	public int getFrom2() 
	{
		return from2;
	}
	
	public String getMessage() 
	{
		return message;
	}
	
	public boolean getBroadcast() 
	{
		return broadcast;
	}
}

File: NetworkS/src/Operation.scala
class Operation (private val gfrom: Dispositive, private val gto: Dispositive, private val gmessage: String, private val gfrom2: Int, private val gto2: Int, private val gbroadcast: Boolean)
{
	if(gbroadcast)
		println("Operazione di broadcast creata da "+gfrom2);
	else
		println("Operazione creata da "+gfrom2+" a "+gto2+".");
    def from = gfrom;
    def to = gto;
	def message = gmessage;
	def from2 = gfrom2;
	def to2 = gto2;
	def broadcast = gbroadcast;
}